reactor-core/src/main/java/reactor/core/publisher/OnNextFailureStrategy.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2011-2017 Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.publisher;
+
+import java.util.function.BiConsumer;
+import java.util.function.BiPredicate;
+import javax.annotation.Nullable;
+
+import org.reactivestreams.Subscription;
+import reactor.core.Exceptions;
+import reactor.util.context.Context;
+
+/**
+ * A strategy to deal with errors that happens during an operator's onNext call, by either
+ * propagating the error or allowing the sequence to continue. This opt-in strategy is
+ * applied by compatible operators through the {@link Operators#onNextFailure(Object, Throwable, Context, Subscription)}
+ * and {@link Operators#onNextPollFailure(Object, Throwable, Context)} methods.
+ * See {@link #STOP}, {@link #RESUME_DROP} and {@link #resume(BiConsumer)} for the
+ * possible strategies.
+ *
+ * @author Simon Baslé
+ */
+public interface OnNextFailureStrategy {
```

Comment:
Is the use of `failure` in this PR instead of `error` deliberate, just wondering because `error` is used everywhere else.